docs/config-merge-logic.md

```diff
@@ -0,0 +1,576 @@
+  was the integer seqno of the update; `"hash"` is the 32-byte, unkeyed BLAKE2b hash of its overall
+  config message; and `"diff"` is as described in the [Config diff updates](#config-diff-updates)
+  section below.
+
+  This list must be ordered by seqno (primary) and hash (secondary) and gives an ability to "replay"
+  recent changes during conflict resolution to merge competing changes, as described in the
+  [Conflict resolution](#conflict-resolution) section, below.
+
+- key `"="` contains the new diff of updates applied in *this* message; it is a dict that mirrors
+  the main application config data, but with placeholder values to indicate changes (rather than
+  duplicating values).
+
+## Config diff updates
+
+A config diff itself (i.e. the third element of a `"<"` tuple, or the `"="` value) is a dict that
+largely mirrors the inner config data structure (i.e.  under the `"&"` outer config key) but that
+only records fields that were specifically changed as part of this update, but *not* include changes
+that were the result of merging other updates.  That is, if a message with seqno 456 adds a new key
+and also resolves a merge conflict which adds a second key then only the first key is included in
+the diff (the second key will already be included in one of the lagged diff entries).
+
+A diff update itself is a dict such that:
+
+- the key indicates the equivalent key inside the config data (`"&"`).
+- a value of `""` (empty string) indicate that a string or integer value was assigned (whether added
+  or changed).
+- a value of `"-"` indicates that a string or integer value was removed from the dict.
+- Note that dicts and sets are never explicitly listed as additions or removals at all: rather they
+  are implicitly removed when empty, and implicit autovivified (if needed) when added to.
+- a dict value indicates that there were changes within an existing sub-dict of the given name.
+  This diff specification recurses for changes within that dict.  Sub-dicts are created or removed
+  as needed (created when the first subkey is added; removed when the last subkey is removed).
+- a value that is a list must be a pair of sublists which indicates that the reference key is a set
+  and that changes were applied to it.  The first element of the sublist is the subset of the list's
+  new values; the second is a subset of removed values.  Both are always required, but one can be
+  empty if there were only additions or only removals.  Similarly to dict handling, removing *all*
+  values implicitly removes the set itself, and adding to a non-existent set autovivifies the set.
+
+# Client update behaviour
+
+When there are multiple conflicting config messages clients resolve according to several rules.
+
+## Ignored updates
+
+Clients ignore updates according to two criteria:
+
+- "within five": a client ignores any update with a seqno value that is not within the last five
```

Comment:
note why it's 5. 

aka: this value controls conflict resolution depth via [insert metric here] so we chose 5 because ass number.

Reply:
This is chosen by the caller now.